rdkit/Chem/Draw/aggCanvas.py

```diff
@@ -137,18 +136,17 @@ def addCanvasText(self, text, pos, font, color=(0, 0, 0), **kwargs):
       subH *= 0.5
       h += supH + subH
       offset = w * pos[2]
-      dPos = [pos[0] - w / 2. + offset, pos[1] - h / 2.]
       if orientation == 'W':
         dPos = [pos[0] - w + offset, pos[1] - h / 2.]
       elif orientation == 'E':
         dPos = [pos[0] + offset, pos[1] - h / 2.]
       else:
-        dPos = [pos[0] - w / 2 + offset, pos[1] - h / 2.]
+        dPos = [pos[0] - w / 2. + offset, pos[1] - h / 2.]
 
       if supH:
         dPos[1] += supH
       for txt, fmt, tw, th in dblocks:
-        tPos = dPos[:]
+        tPos = dPos.copy()
```

Comment:
Keep the old slice. If that worked before it means that a copy is not required.

Reply:
`dPos[:]` means you are making a copy, so these you are completely identical. In new Python , list.copy() is faster
https://stackoverflow.com/questions/2612802/list-changes-unexpectedly-after-assignment-why-is-this-and-how-can-i-prevent-it
https://www.python.org/dev/peps/pep-0590/

Reply:
Ok - good to know.